File: work-order-vue-app/src/types/work-order.ts
export type WorkOrderStatus = 'Pending' | 'InProgress' | 'Completed' | 'Cancelled'

export interface WorkOrder {
  id: number
  title: string
  description: string | null
  status: WorkOrderStatus
  assignedTo: string
  createdAt: string
  dueDate: string | null
  completedAt: string | null
  activities: Activity[]
}

export interface Activity {
  id: number
  workOrderId: number
  action: string
  details: string | null
  performedBy: string
  timestamp: string
}

export interface WorkOrderListResponse {
  items: WorkOrder[]
  page: number
  pageSize: number
  totalCount: number
}

export interface CreateWorkOrderRequest {
  title: string
  description?: string
  assignedTo: string
  dueDate?: string
}

export interface UpdateWorkOrderStatusRequest {
  status: WorkOrderStatus
}

File: work-order-api/Models/WorkOrder.cs
namespace WorkOrderApi.Models;

public class WorkOrder
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public WorkOrderStatus Status { get; set; } = WorkOrderStatus.Pending;

    public string AssignedTo { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public DateTime? DueDate { get; set; }

    public DateTime? CompletedAt { get; set; }

    public ICollection<Activity> Activities { get; set; } = new List<Activity>();
}
